Brokerages/Bitfinex/BitfinexBrokerage.cs

```diff
@@ -0,0 +1,449 @@
+                    Exchange = _exchange,
+                    Side = order.Quantity > 0 ? buy : sell
+                };
+                var response = _restClient.CancelReplaceOrder(post);
+                if (response.OrderId == 0)
+                {
+                    hasFaulted = true;
+                    break;
+                }
+            }
+
+            if (hasFaulted)
+            {
+                return false;
+            }
+
+            UpdateCachedOpenOrder(order.Id, order);
+            return true;
+        }
+
+        /// <summary>
+        /// Cancel an existing order
+        /// </summary>
+        /// <param name="order"></param>
+        /// <returns></returns>
+        public override bool CancelOrder(Orders.Order order)
+        {
+            try
+            {
+                Log.Trace("BitfinexBrokerage.CancelOrder(): Symbol: " + order.Symbol.Value + " Quantity: " + order.Quantity);
+
+                foreach (var id in order.BrokerId)
+                {
+                    var response = _restClient.CancelOrder(int.Parse(id));
+                    if (response.Id > 0)
+                    {
+                        Order cached;
+                        this.CachedOrderIDs.TryRemove(order.Id, out cached);
+                        const int orderFee = 0;
+                        OnOrderEvent(new OrderEvent(order, DateTime.UtcNow, orderFee, "Bitfinex Cancel Order Event") { Status = OrderStatus.Canceled });
+                    }
+                    else
+                    {
+                        return false;
+                    }
+
+                }
+            }
+            catch (Exception err)
+            {
+                Log.Error("CancelOrder(): OrderID: " + order.Id + " - " + err);
+                return false;
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Setup ticker polling
+        /// </summary>
+        public override void Connect()
+        {
+            _tickerToken = new CancellationTokenSource();
+        }
+
+        /// <summary>
+        /// Cancel ticker polling
+        /// </summary>
+        public override void Disconnect()
+        {
+            if (this._tickerToken != null)
+            {
+                this._tickerToken.Cancel();
+            }
+        }
+
+        private List<BitfinexOrder> GetOpenBitfinexOrders()
+        {
+            var list = new List<BitfinexOrder>();
+
+            try
+            {
+                var response = _restClient.GetActiveOrders();
+                if (response != null)
+                {
+                    foreach (var item in response)
+                    {
+                        list.Add(new BitfinexOrder
+                        {
+                            Quantity = Convert.ToInt32(decimal.Parse(item.OriginalAmount) * ScaleFactor),
+                            BrokerId = new List<string> { item.Id.ToString() },
```

Comment:
Are all bitfinex orders market orders? I would have expected them to minimally also support limit orders.